### ISSUE TITLE
Move github pages to doxygen.nubots.net

### ISSUE THREAD

**Issue by CMurtagh-LGTM** [NUbots/NUbots#877]
Currently we host our doxygen output on nubots.github.io/NUbots Trent suggested putting it on nubots.net as this is were we have NUbook.
https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site

**Comment by JosephusPaye** [comment]
We currently have the nubots.net domain managed in Netlify. Let me see about adding the subdomain.

**Comment by LachlanCourt** [comment]
I would suggest the url just be `docs.nubots.net` just so that it's one less thing to change if we decide to move away from doxygen in the future

**Comment by JosephusPaye** [comment]
Good point. Other options could be `codedocs.` or `api.` perhaps? Thinking of someone getting confused with `docs.` not being going to NUbook as NUbook is technically also "docs".

**Comment by JosephusPaye** [comment]
It's done: https://codedocs.nubots.net/
